tests/test_main.py

```diff
@@ -0,0 +1,209 @@
+from main import NOTIFY_BASE_URL, send_email
+
+url = f"{NOTIFY_BASE_URL}/notifications/email"
+
+
+def test_get_not_allowed():
+    request = Mock(method="GET")
+    response = send_email(request)
+    assert response == ("Method not allowed", 405)
+
+
+def test_missing_data_returns_422():
+    request = Mock(method="POST", json={})
+    response = send_email(request)
+    assert response == ("Missing notification request data", 422)
+
+
+def test_invalid_service_id_raises_invalid_notify_key_error():
+    random_string = "".join(random.choice(string.printable) for i in range(87))
+    with mock.patch.dict(os.environ, {"NOTIFY_API_KEY": random_string}):
+        with pytest.raises(InvalidNotifyKeyError):
+            reload(main)
+
+
+def test_invalid_api_key_raises_invalid_notify_key_error():
+    random_string = "".join(random.choice(string.printable) for i in range(37))
+    uuid_string = str(uuid4())
+    with mock.patch.dict(os.environ, {"NOTIFY_API_KEY": uuid_string + random_string}):
+        with pytest.raises(InvalidNotifyKeyError):
+            reload(main)
+
+
+@responses.activate
+def test_notify_response_error_returns_correctly():
+    request = Mock(
+        method="POST",
+        json={
+            "payload": {
+                "fulfilmentRequest": {
+                    "email_address": "test@example.com",
+                    "personalisation": {"address": "test address"},
+                    "form_type": "H",
+                    "language_code": "en",
+                    "region_code": "GB-ENG",
+                }
+            },
+        },
+    )
+    responses.add(responses.POST, url, json={"errors": "403"}, status=403)
+    response = send_email(request)
+    assert response == ("Notify request failed", 403)
+
+
+@responses.activate
+def test_notify_response_no_content_204():
+    request = Mock(
+        method="POST",
+        json={
+            "payload": {
+                "fulfilmentRequest": {
+                    "email_address": "test@example.com",
+                    "personalisation": {"address": "test address"},
+                    "form_type": "H",
+                    "language_code": "en",
+                    "region_code": "GB-ENG",
+                },
+            },
+        },
+    )
+    responses.add(responses.POST, url, json={}, status=204)
+    response = send_email(request)
+    assert response == ("No content", 204)
+
+
+@responses.activate
+def test_notify_response_json_decode_error():
+    request = Mock(
+        method="POST",
+        json={
+            "payload": {
+                "fulfilmentRequest": {
+                    "email_address": "test@example.com",
+                    "personalisation": {"address": "test address"},
+                    "form_type": "H",
+                    "language_code": "en",
+                    "region_code": "GB-ENG",
+                },
+            },
+        },
+    )
+    responses.add(responses.POST, url, status=200)
```

Comment:
I think it's worth adding a comment here to make it clear that this isn't a response that should ever be expected to be returned by Notify.